src/acquisition/flusurv/flusurv_update.py

```diff
@@ -175,18 +321,28 @@ def main():
     # fmt: on
     args = parser.parse_args()
 
+    metadata = flusurv.fetch_flusurv_metadata()
+
     # scrape current issue from the main page
-    issue = flusurv.get_current_issue()
+    issue = flusurv.get_current_issue(metadata)
     print(f"current issue: {int(issue)}")
 
+    # Ignore seasons with all dates older than one year
+    seasonids = {
+        season_blob["seasonid"] for season_blob in metadata["seasons"]
+        if delta_epiweeks(flusurv.mmwrid_to_epiweek(season_blob["endweek"]), issue) < max_age_to_consider_weeks
+    }
+
     # fetch flusurv data
     if args.location == "all":
         # all locations
-        for location in flusurv.location_codes.keys():
-            update(issue, location, args.test)
+        for location in flusurv.location_to_code.keys():
+            update(issue, location, seasonids, metadata, args.test)
     else:
         # single location
-        update(issue, args.location, args.test)
+        assert args.location in flusurv.location_to_code.keys(), \
+            f"Requested location {args.location} not available"
```

Comment:
nice check!  how about raising a proper `Exception` instead of the `assert`?  also, the `.keys()` call after `in` isnt necessary.
```suggestion
        assert args.location in flusurv.location_to_code, \
            f"Requested location {args.location} not available"
```

Reply:
Going to leave `.keys()` for clarity.